File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.16-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_16() {
    let chars = ['a', 'b', 'c', 'b', 'd'];
    let ind = 0;

    let row = chars.length;
    let column = 5;

    for (let y = row - 1; y >= 0; y--) {
        for (let x = 0; x < column; x++) {
            if (x <= y) {
                log(chars[ind]);
            } else {
                log(' ');
            }
            ind = ind + 1;
        }

        ind = 0;

        console.log();
    }
}

pattern_16();

File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.16-pattern.py
from utils.clear import clear


def pattern_16():
    row = 5
    column = 5

    chars = ['a', 'b', 'c', 'd', 'e']

    for i in range(row - 1, -1, -1):
        for j in range(0, column):
            if j <= i:
                print(chars[j], end="")
            elif j > i:
                print(" ", end="")

        print()

clear()
pattern_16()
